lib/sublayer/cli/commands/generators/sublayer_generator_generator.rb

```diff
@@ -0,0 +1,96 @@
+class SublayerGeneratorGenerator < Sublayer::Generators::Base
+  llm_output_adapter type: :named_strings,
+    name: "sublayer_generator",
+    description: "The new Sublayer generator code based on the description",
+    attributes: [
+      { name: "code", description: "The code of the generated Sublayer generator" },
+      { name: "filename", description: "The filename of the generated Sublayer generator camel cased and with a .rb extension" }
+    ]
+
+  def initialize(description:)
+    @description = description
+  end
+
+  def generate
+    super
+  end
+
+  def prompt
+    <<-PROMPT
+    You are an expert ruby programmer and and great at repurposing code examples to use for new situations.
+
+    A Sublayer generator is a class that acts as an abstraction for sending a request to an LLM and getting structured data back.
+
+    An example Sublayer generator is:
+    <example_generator>
+    #{example_generator}
+    </example_generator>
+
+    And it is the generator we're currently using for taking in a description from a user and generating a new Sublayer generator.
+
+    All Sublayer::Generators inherit from Sublayer::Generators::Base and have a generate method that simply calls super for the user to use and modify for their uses.
+
+    the llm_output_adapter directive is used to instruct the LLM on what structure of output to generate. In the example we're using type: :single_string which takes a name and description as arguments.
+
+    the other available options and their example usage is:
+    llm_output_adapter type: :list_of_strings,
+      name: "suggestions",
+      description: "List of keyword suggestions"
+
+    llm_output_adapter type: :single_integer,
+      name: "four_digit_passcode",
+      description: "an uncommon and difficult to guess four digit passcode"
+
+    llm_output_adapter type: :list_of_named_strings,
```

Comment:
gonna open an issue for this.

one idea here is that we can create a very flexible output adapter that is as amorphic as (direct translation of) the json shema (essential takes type as a named param and can do so recursively) 

it could serve as both a fallback adapter as well as a simple adapter for this kind of use case (prompts)

Reply:
#101